File: Algorithms/Graphs/BFS.java
package defaultt;
import java.util.*;
public class Graph {
	
	public static void printBFSHelper(int edges[][],int sv,boolean visited[])
	{
		Queue<Integer> q=new LinkedList<>();
		q.add(sv);
		visited[sv]=true;
		int n=edges.length;
		while(!q.isEmpty())
		{
			int front;
			try
			{
				front=q.poll();	
			}
			catch (Exception e)
			{
				return ;
			}
			System.out.println(front);
			for(int i=0;i<n;i++)
			{
				if(edges[front][i]==1 && !visited[i])
				{
					q.add(i);
					visited[i]=true;
				}
			}
		}
		
	}
	public static void printBFS(int edges[][])
	{
		boolean visited[]=new boolean[edges.length];
		for(int i=0;i<edges.length;i++)
		{
			if(!visited[i])
			printBFSHelper(edges,i,visited);
		}
		
	}
	
	public static void main(String[] args)
	{
		int n;
		int e;
		Scanner s=new Scanner(System.in);
		n=s.nextInt();
		e=s.nextInt();
		int edges[][]=new int[n][n];
		//wherever there is edge we keep 1
		for(int i=0;i<e;i++)
		{
			int fv=s.nextInt();  //final vertex
			int sv=s.nextInt();  //starting vertex
			edges[fv][sv]=1;
			edges[sv][fv]=1;
		}
		System.out.println("BFS");
		printBFS(edges);
	}

}

File: Algorithms/bfs.cpp
#include<queue>
using namespace std;

void printBFS(int** edges, int n,int sv, bool *visited)
{
    queue<int> q;
    q.push(sv);
    //visited[0] = true;
    while(q.size() != 0)
    {
        visited[q.front()] = true;
        for(int i = 0 ; i < n; i++)
        {
            if(i == q.front())
            {
                continue;
            }
            if(edges[q.front()][i] == 1)
            {
                if(visited[i] == true)
                {
                    continue;
                }
                q.push(i);
                 visited[i] = true;
            }
        }
        cout << q.front() <<" ";
        q.pop();
    }
    for(int i = 0; i < n; i++)
    {
        if(visited[i] == false)
        {
            printBFS(edges,n,i,visited);
        }
    }
}

int main() {
    int V, E;
    cin >> V >> E;
    int** edges = new int*[V];
    for(int i = 0; i < V; i++)
    {
        edges[i] = new int [V];
        for(int j = 0; j < V; j++)
        {
            edges[i][j] = 0;
        }
    }
    for(int i = 0; i < E; i++)
    {
        int a,b;
        cin >> a >> b;
        edges[a][b] = 1;
        edges[b][a] = 1;
    }
    bool visited[V] = {false};
    printBFS(edges,V,0,visited);
  return 0;
}
